src/braket/experimental/autoqasm/program/program.py

```diff
@@ -447,6 +448,74 @@ def push_oqpy_program(self, oqpy_program: oqpy.Program) -> None:
         finally:
             self._oqpy_program_stack.pop()
 
+    @contextlib.contextmanager
+    def if_block(self, condition: Any) -> None:
+        """Sets the program conversion context into an if block context.
+
+        Args:
+            condition (Any): The condition of the if block.
+        """
+        oqpy_program = self.get_oqpy_program()
+        current_in_global_scope = self.at_root_scope
+        try:
+            self.at_root_scope = False
```

Comment:
Ah, I didn't understand what this variable meant until I hit this line. It's a little confusing, but I see now that `main` top level scope is root. Inside an if/for/while, it's no longer top level scope of main, so no longer root. Is the top level scope of a subroutine also the root scope?

Could you add a documentation line added below line 202?

Reply:
Yes, this means at the top-level scope of a function (either main or a subroutine). I'll rename this to `at_function_root_scope` to be a little bit clearer, and add a comment explaining.